fh-messaging/lib/fh_metrics_man.js

```diff
@@ -197,24 +197,67 @@ MetricsManager.prototype.deleteMetricsData = function(topic, date, callback) {
   // if multiple message managers pointed at the same config are running at the
   // same time
   var msgConfig = JSON.parse(JSON.stringify(self.mConfig));
-  var collectionName = topic + "_" + helpers.toYYYYMMDD(date);
-
   messageMan = new fhmsg.Messaging(msgConfig, self.mLogger);
   self.messageMans.push(messageMan);
+
   messageMan.database.on('tearUp', function() {
-    messageMan.database.db.dropCollection(collectionName, callback);
-    //remove all the data that is older than the x number of days from the intermediate tables
-    async.each(['transactionsdesttemp', 'transactionsgeotemp'], function(tempCol, cb) {
-      var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
-      var beforeDate = new Date(date.getFullYear(), date.getMonth(), date.getDate() - daysToCalculate);
-      var query = {"_id.ts":{$lt: beforeDate.getTime()}};
-      messageMan.database.db.collection(tempCol, function(err, collection) {
+    // Pruning for fhact
+    var daysToCalculate = self.mConfig.agenda.jobs.metrics_rollup_job.daysToKeep || 31;
+    var pruning_date = new Date().getTime() - ( daysToCalculate * 24 * 60 * 60 * 1000 );
+    var prune_collection_names = [];
+
+    // Retrieve the list of collections from the database
+    messageMan.database.db.listCollections({}).toArray( function(err, items) {
+      if (err) {
+        self.mLogger.error("listCollections error: " + err);
+        callback(err);
+      }
+
+      // Generate list of prunable collections
+      for (var i=0; i < items.length; i++) {
+        if (items[i].name.includes(topic +"_")) {
+          self.mLogger.debug('Found ' + topic + ' collection: ' + items[i].name);
+          // construct Date from the "fhact_YYYYMMDD" name
+          var topic_date_string = items[i].name.substring((topic + '_').length, items[i].name.length);
+          // helper method to construct date
+          var collection_date = helpers.toDateFromYYYYMMDD(topic_date_string);
+          // check if this date is older than the number of days to prune, append to list if so
+          self.mLogger.debug('Check collection_date.getTime(): ' + collection_date.getTime());
+          self.mLogger.debug('Check pruning_date: ' + pruning_date);
+          self.mLogger.debug('Check if date is prunable ' + (collection_date.getTime() < parseInt(pruning_date)));
+          if ( collection_date.getTime() < parseInt(pruning_date) ) {
```

Comment:
It shows not be in the correct place. Why not just select the collections with this conditional?

Reply:
Hmm I might need a little help formatting that correctly, if you can think of a better way to make the call and get the list of collections which meet the criteria for pruning! From the API i couldn't figure out a way of getting the collections if you didn't already know the name of the collection.